### PR TITLE
Miscellaenous improvements to std.array unit tests

### PR DESCRIPTION
**Pull request by dukc**

### CODE REVIEW

File: std/array.d [hunk 1]
```diff
@@ -247,12 +245,20 @@ version (StdUnittest)
     assert(nAlive == 3);
 
     // No good way to ensure the GC frees this, just call the lifetime function
-    // directly. If delete wasn't deprecated, this is what delete would do.
-    _d_delarray_t(cast(void[]*)&arr, typeid(S));
+    // directly.
+    __delete(arr);
```
Comment (atilaneves) [inline std/array.d:249]: Why not `destroy`?

File: std/array.d [hunk 2]
```diff
@@ -421,14 +427,6 @@ if (isAutodecodableString!String)
     });
 }
 
-@safe unittest
```
Comment (atilaneves) [inline std/array.d:424]: Why was this moved?

**Review by RazvanN7 — APPROVED** [review]

File: std/array.d (hunk 2 above)
Comment (dukc) [reply std/array.d:424]: It was under wrong public function

File: std/array.d (hunk 1 above)
Comment (dukc) [reply std/array.d:249]: It failed the test when I tried. I think `destory` finalizes only the array itself, not it's contents.

**Comment by RazvanN7** [comment]
@dukc The documentation tester seems to have a spurious failure. Please push again your branch to restart it.

File: std/array.d (hunk 1 above)
Comment (n8sh) [reply std/array.d:249]: [Yeah it falls into this case:](https://github.com/dlang/druntime/blob/8d80dec1f6fc2a69a3f26fdfbc704079f8a0d661/src/object.d#L4298-L4304)
```d
/// ditto
void destroy(bool initialize = true, T)(ref T obj)
    if (!is(T == struct) && !is(T == interface) && !is(T == class) && !__traits(isStaticArray, T))
{
    static if (initialize)
        obj = T.init;
}
```

**Review by atilaneves — APPROVED** [review]
